code/modules/reagents/chemistry/machinery/chem_heater.dm

```diff
@@ -34,12 +34,31 @@
 		QDEL_NULL(beaker)
 	return ..()
 
-/obj/machinery/chem_heater/Exited(atom/movable/gone, direction)
-	. = ..()
-	if(gone == beaker)
-		UnregisterSignal(beaker.reagents, COMSIG_REAGENTS_REACTION_STEP)
-		beaker = null
-		update_appearance()
+
+/obj/machinery/chem_heater/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	if(isnull(held_item) || (held_item.item_flags & ABSTRACT) || (held_item.flags_1 & HOLOGRAM_1))
+		return NONE
+
+	if(!QDELETED(beaker))
+		if(istype(held_item, /obj/item/reagent_containers/dropper) || istype(held_item, /obj/item/reagent_containers/syringe))
+			context[SCREENTIP_CONTEXT_LMB] = "Inject"
+			return CONTEXTUAL_SCREENTIP_SET
+		if(is_reagent_container(held_item)  && held_item.is_open_container())
+			context[SCREENTIP_CONTEXT_LMB] = "Replace beaker"
+			return CONTEXTUAL_SCREENTIP_SET
+	else if(is_reagent_container(held_item)  && held_item.is_open_container())
```

Comment:
you really mean to tell me `is_open_container` is defined on /obj/item

Reply:
Not on obj but on atom
https://github.com/tgstation/tgstation/blob/d867d56af2e9985e5990b6a5cc96a455fcbd91f4/code/game/atom/_atom.dm#L381